src/Microsoft.Health.Dicom.Core/Features/ExtendedQueryTag/ExtendedQueryTagErrorsService.cs

```diff
@@ -0,0 +1,50 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Extensions;
+using Microsoft.Health.Dicom.Core.Features.Common;
+using Microsoft.Health.Dicom.Core.Messages.ExtendedQueryTag;
+
+namespace Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag
+{
+    public class ExtendedQueryTagErrorsService : IExtendedQueryTagErrorsService
+    {
+        private readonly IExtendedQueryTagErrorStore _extendedQueryTagErrorStore;
+        private readonly IDicomTagParser _dicomTagParser;
+
+        public ExtendedQueryTagErrorsService(IExtendedQueryTagErrorStore extendedQueryTagStore, IDicomTagParser dicomTagParser)
+        {
+            _extendedQueryTagErrorStore = EnsureArg.IsNotNull(extendedQueryTagStore, nameof(extendedQueryTagStore));
+            _dicomTagParser = EnsureArg.IsNotNull(dicomTagParser, nameof(dicomTagParser));
+        }
+
+        public async Task<GetExtendedQueryTagErrorsResponse> GetExtendedQueryTagErrorsAsync(string tagPath, CancellationToken cancellationToken = default)
```

Comment:
I am little confused. Should'nt you be querying on all the errors for a specific re-indexing run?
How is this supported?
I thought the Job ID is the source of the errors and would have some reference to it.

Reply:
Peng Will and I thought that since the errors are Tag errors, it is more intuitive for the users to query errors associated with a tag than querying by operation.